### PR TITLE
Fix: update initialization of ionizationMode

### PR DESCRIPTION
**Pull request by IpankajI**
ionizationMode was updated with the index value not with the
actual value

### CODE REVIEW

File: src/core/libmaven/mavenparameters.cpp [hunk 1]
```diff
@@ -232,8 +232,15 @@ void MavenParameters::setOptionsDialogSettings(const char* key, const char* valu
...
     mavenSettings[const_cast<char*>(key)] = const_cast<char*>(value);
 
-    if(strcmp(key, "ionizationMode") == 0)
-        ionizationMode = atof(value);
+    if(strcmp(key, "ionizationMode") == 0){
+        int indexOfIonizationMode=*value-'0';
+        /**
+         * map index of ionization-mode to value (-1,0,1) of ionizationMode
+         */
+        if(indexOfIonizationMode==1) ionizationMode=0;
```
Comment (shubhra-agrawal) [inline src/core/libmaven/mavenparameters.cpp]: MainWindow::setTotalCharge() does the same thing. Check that function out.

Comment (rish9511) [reply src/core/libmaven/mavenparameters.cpp]: @shubhra-agrawal  we still need to do it here. if we don't then, every call to this function will change the ionizationMode to hold the index and not the actual value

Comment (shubhra-agrawal) [reply src/core/libmaven/mavenparameters.cpp]: @rish9511 Yes, we need to do it here. I merely suggested using the already defined function here if possible. Short answer is, we cannot. So this is fine.

File: src/core/libmaven/mavenparameters.cpp [hunk 2]
```diff
@@ -232,8 +232,15 @@ void MavenParameters::setOptionsDialogSettings(const char* key, const char* valu
 {
     mavenSettings[const_cast<char*>(key)] = const_cast<char*>(value);
 
-    if(strcmp(key, "ionizationMode") == 0)
-        ionizationMode = atof(value);
+    if(strcmp(key, "ionizationMode") == 0){
+        int indexOfIonizationMode=*value-'0';
```
Comment (shubhra-agrawal) [inline src/core/libmaven/mavenparameters.cpp]: @IpankajI Suggest using atoi() or atod() here for consistency.

File: src/core/libmaven/mavenparameters.cpp [hunk 3]
```diff
@@ -232,8 +232,17 @@ void MavenParameters::setOptionsDialogSettings(const char* key, const char* valu
...
     mavenSettings[const_cast<char*>(key)] = const_cast<char*>(value);
 
-    if(strcmp(key, "ionizationMode") == 0)
-        ionizationMode = atof(value);
+    if(strcmp(key, "ionizationMode") == 0){
+        int indexOfIonizationMode=atoi(value);
+        /**
+         * map index of ionization-mode to value (-1,0,1) of ionizationMode
+         */
+        if(indexOfIonizationMode==1) ionizationMode=0;
```
Comment (shubhra-agrawal) [inline src/core/libmaven/mavenparameters.cpp]: @IpankajI Shouldn't the indices be 0,1 and 2?

Comment (IpankajI) [reply src/core/libmaven/mavenparameters.cpp]: @shubhra-agrawal 0 is for Auto-Detect, rest is tested.

Comment (shubhra-agrawal) [reply src/core/libmaven/mavenparameters.cpp]: @IpankajI I was confusing this drop-down with the one on main window. My bad.

**Pull request merged**